libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/components/customManifestsValidationSchema.tsx

```diff
@@ -13,45 +12,40 @@ const INCORRECT_FILENAME = 'Must have a yaml, yml or json extension and can not
 const UNIQUE_FOLDER_FILENAME =
   'Ensure unique file names within each folder to avoid conflicts and errors.';
 
-export const getUniqueValidationSchema = <FormValues,>(
-  uniqueStringArrayExtractor: UniqueStringArrayExtractor<FormValues>,
-) => {
-  return Yup.string().test('unique', UNIQUE_FOLDER_FILENAME, function (value: string) {
-    const context = this.options.context as Yup.TestContext & { values?: FormValues };
+export const getUniqueValidationSchema = Yup.string().test(
+  'unique',
+  UNIQUE_FOLDER_FILENAME,
+  function (_value: string) {
+    const context = this.options.context as Yup.TestContext & { values?: ManifestFormData };
+    const { filename, folder } = this.parent as CustomManifestValues;
+
     if (!context || !context.values) {
       return this.createError({
         message: 'Unexpected error: Yup test context should contain form values',
       });
     }
 
-    const values = uniqueStringArrayExtractor(context.values, this, value);
-
-    const setValues = new Set(values);
-
-    if (!values) {
-      return this.createError({
-        message: 'Unexpected error: Failed to get values to test uniqueness',
-      });
-    }
-    return values.length === setValues.size;
-  });
-};
+    const filePath = `${folder}_${filename}`;
+    const values = context.values.manifests.map(
+      (manifest) => `${manifest.folder}_${manifest.filename}`,
```

Comment:
I think that this has changed in #2277

Reply:
Yes, this has been changed in #2277

Reply:
Don't worry, I'll do a rebase. :slightly_smiling_face:

Reply:
@jgyselov resolve the conflicts